server/sonar-server/src/main/java/org/sonar/server/project/ws/CreateAction.java

```diff
@@ -0,0 +1,124 @@
+import static org.sonar.core.permission.GlobalPermissions.PROVISIONING;
+import static org.sonar.server.component.NewComponent.newComponentBuilder;
+import static org.sonar.server.ws.KeyExamples.KEY_PROJECT_EXAMPLE_001;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+import static org.sonarqube.ws.client.project.ProjectsWsParameters.ACTION_CREATE;
+import static org.sonarqube.ws.client.project.ProjectsWsParameters.PARAM_BRANCH;
+import static org.sonarqube.ws.client.project.ProjectsWsParameters.PARAM_NAME;
+
+public class CreateAction implements ProjectsWsAction {
+
+  public static final String PARAM_ID = "id";
+  public static final String PARAM_KEY = "key";
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final ComponentUpdater componentUpdater;
+  private final DefaultOrganizationProvider defaultOrganizationProvider;
+
+  public CreateAction(DbClient dbClient, UserSession userSession, ComponentUpdater componentUpdater, DefaultOrganizationProvider defaultOrganizationProvider) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.componentUpdater = componentUpdater;
+    this.defaultOrganizationProvider = defaultOrganizationProvider;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION_CREATE)
+      .setDescription("Create a project.<br/>" +
+        "Requires 'Create Projects' permission")
+      .setSince("4.0")
+      .setPost(true)
+      .setResponseExample(getClass().getResource("create-example.json"))
+      .setHandler(this);
+
+    action.createParam(ProjectsWsParameters.PARAM_KEY)
+      .setDescription("Key of the project")
+      .setRequired(true)
+      .setExampleValue(KEY_PROJECT_EXAMPLE_001);
+
+    action.createParam(PARAM_NAME)
+      .setDescription("Name of the project")
+      .setRequired(true)
+      .setExampleValue("SonarQube");
+
+    action.createParam(PARAM_BRANCH)
+      .setDescription("SCM Branch of the project. The key of the project will become key:branch, for instance 'SonarQube:branch-5.0'")
+      .setExampleValue("branch-5.0");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    userSession.checkPermission(PROVISIONING);
+    CreateRequest createRequest = toCreateRequest(request);
+    writeProtobuf(doHandle(createRequest), request, response);
+  }
+
+  private CreateWsResponse doHandle(CreateRequest request) {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      ComponentDto componentDto = componentUpdater.create(dbSession, newComponentBuilder()
+        .setOrganizationUuid(defaultOrganizationProvider.get().getUuid())
+        .setKey(request.getKey())
+        .setName(request.getName())
+        .setBranch(request.getBranch())
+        .setQualifier(PROJECT)
+        .build(),
+        userSession.isLoggedIn() ? userSession.getUserId().longValue() : null);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'getUserId' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)